custom_components/lennoxs30/sensor.py

```diff
@@ -242,3 +248,55 @@ def unit_of_measurement(self):
     @property
     def device_class(self):
         return DEVICE_CLASS_HUMIDITY
+
+
+class S30InverterPowerSensor(SensorEntity):
+    """Class for Lennox S30 inverter power."""
+
+    def __init__(self, hass: HomeAssistant, manager: Manager, system: lennox_system):
+        self._hass = hass
+        self._manager = manager
+        self._system = system
+        self._system.registerOnUpdateCallback(
+            self.update_callback,
+            ["diagInverterInputVoltage", "diagInverterInputCurrent"]
+        )
+        self._myname = self._system.name + "_inverter_energy"
+
+    def update_callback(self):
+        _LOGGER.info(f"update_callback myname [{self._myname}]")
+        self.schedule_update_ha_state()
+
+    @property
+    def unique_id(self) -> str:
+        # HA fails with dashes in IDs
+        return (self._system.unique_id() + "_IE").replace("-", "")
+
+    def update(self):
+        """Update data from the thermostat API."""
+        return True
+
+    @property
+    def should_poll(self):
+        """No polling needed."""
+        return False
+
+    @property
+    def name(self):
+        return self._myname
+
+    @property
+    def state(self):
+        try:
+            return float(self._system.diagInverterInputVoltage) * float(self._system.diagInverterInputCurrent)
```

Comment:
Should we try to control the precision here.  For example, reporting 70.0023 watts then 70.0024 watts would not be ideal.

Rounding to the nearest Watt may be sufficient?  1/10 watt?

Reply:
Very good point, integer-precision is probably plenty for this range (70-1500W from what I've seen so far)